src/LanguageServer/Protocol/Handler/Symbols/DocumentSymbolsHandler.cs

```diff
@@ -25,7 +25,7 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// </summary>
     [ExportCSharpVisualBasicStatelessLspService(typeof(DocumentSymbolsHandler)), Shared]
     [Method(Methods.TextDocumentDocumentSymbolName)]
-    internal sealed class DocumentSymbolsHandler : ILspServiceDocumentRequestHandler<RoslynDocumentSymbolParams, object[]>
+    internal sealed class DocumentSymbolsHandler : ILspServiceDocumentRequestHandler<RoslynDocumentSymbolParams, SumType<RoslynDocumentSymbol[], SymbolInformation[]>>
```

Comment:
could this be `SumType<DocumentSymbol[], SymbolInformation[]>` (and return RoslynDocumentSymbol) or does that not work?

Reply:
That's what I had at first, but the Document Outline tests failed. Guessing it is missing a converter, but it wasn't clear to me why that uses LSP at all, so didn't want to poke the bear :)

Reply:
I would guess the document outline tests just need to request the base type, and then cast the result it gets back here https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/CSharp/Test/DocumentOutline/DocumentOutlineTestsBase.cs#L89

Reply:
It's an actual serialization issue, the `Glyph` property is just lost. I could probably add a call to `AddOrReplaceConverter<DocumentSymbol, RoslynDocumentSymbol>();` [here](https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Protocol/Internal/Converters/VSInternalExtensionUtilities.cs#L31) though if you like?

Reply:
if that works, fine with me.  otherwise OK with this too

Reply:
I added the converter, seems to work fine. It's cleaner on the Razor side too, which is nice.

Reply:
lol, that caused a different set of test failures 🤦‍♂️